File: app/src/test/java/linusfessler/alarmtiles/TimeOfDayFormatterTest.java
package linusfessler.alarmtiles;

import org.junit.Assert;
import org.junit.Test;

import linusfessler.alarmtiles.shared.TimeOfDayFormatter;

public class TimeOfDayFormatterTest {

    private final TimeOfDayFormatter timeOfDayFormatter12 = new TimeOfDayFormatter(false);
    private final TimeOfDayFormatter timeOfDayFormatter24 = new TimeOfDayFormatter(true);

    @Test
    public void format24HoursFormatAm() {
        // GIVEN
        final int hourOfDay = 4;
        final int minutesOfHour = 45;
        final String expected = "4:45";
        // WHEN
        final String actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void format24HoursFormatPm() {
        // GIVEN
        final int hourOfDay = 16;
        final int minutesOfHour = 45;
        final String expected = "16:45";
        // WHEN
        final String actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void format12HoursFormatAm() {
        // GIVEN
        final int hourOfDay = 4;
        final int minutesOfHour = 45;
        final String expected = "4:45 AM";
        // WHEN
        final String actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void format12HoursFormatPm() {
        // GIVEN
        final int hourOfDay = 16;
        final int minutesOfHour = 45;
        final String expected = "4:45 PM";
        // WHEN
        final String actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void format24HoursFormatPreviousDay() {
        // GIVEN
        final int hourOfDay = -1;
        final int minutesOfHour = -1;
        final String expected = "22:59";
        // WHEN
        final String actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void format24HoursFormatNextDay() {
        // GIVEN
        final int hourOfDay = 25;
        final int minutesOfHour = 1;
        final String expected = "1:01";
        // WHEN
        final String actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void format12HoursFormatMidnight() {
        // GIVEN
        final int hourOfDay = 0;
        final int minutesOfHour = 0;
        final String expected = "12:00 AM";
        // WHEN
        final String actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }

    @Test
    public void format12HoursFormatMidday() {
        // GIVEN
        final int hourOfDay = 12;
        final int minutesOfHour = 0;
        final String expected = "12:00 PM";
        // WHEN
        final String actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour);
        // THEN
        Assert.assertEquals(expected, actual);
    }
}

File: app/src/test/java/linusfessler/alarmtiles/TimeOfDayFormatterTest.kt
package linusfessler.alarmtiles

import linusfessler.alarmtiles.shared.TimeOfDayFormatter
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class TimeOfDayFormatterTest {
    private val timeOfDayFormatter12 = TimeOfDayFormatter(false)
    private val timeOfDayFormatter24 = TimeOfDayFormatter(true)

    @Test
    fun format24HoursFormatAm() {
        // GIVEN
        val hourOfDay = 4
        val minutesOfHour = 45
        val expected = "4:45"
        // WHEN
        val actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun format24HoursFormatPm() {
        // GIVEN
        val hourOfDay = 16
        val minutesOfHour = 45
        val expected = "16:45"
        // WHEN
        val actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun format12HoursFormatAm() {
        // GIVEN
        val hourOfDay = 4
        val minutesOfHour = 45
        val expected = "4:45 AM"
        // WHEN
        val actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun format12HoursFormatPm() {
        // GIVEN
        val hourOfDay = 16
        val minutesOfHour = 45
        val expected = "4:45 PM"
        // WHEN
        val actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun format24HoursFormatPreviousDay() {
        // GIVEN
        val hourOfDay = -1
        val minutesOfHour = -1
        val expected = "22:59"
        // WHEN
        val actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun format24HoursFormatNextDay() {
        // GIVEN
        val hourOfDay = 25
        val minutesOfHour = 1
        val expected = "1:01"
        // WHEN
        val actual = timeOfDayFormatter24.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun format12HoursFormatMidnight() {
        // GIVEN
        val hourOfDay = 0
        val minutesOfHour = 0
        val expected = "12:00 AM"
        // WHEN
        val actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }

    @Test
    fun format12HoursFormatMidday() {
        // GIVEN
        val hourOfDay = 12
        val minutesOfHour = 0
        val expected = "12:00 PM"
        // WHEN
        val actual = timeOfDayFormatter12.format(hourOfDay, minutesOfHour)
        // THEN
        Assertions.assertEquals(expected, actual)
    }
}
